src/Symfony/Component/Notifier/Recipient/Recipient.php

```diff
@@ -11,18 +11,27 @@
 
 namespace Symfony\Component\Notifier\Recipient;
 
+use Symfony\Component\Notifier\Exception\InvalidArgumentException;
+
 /**
  * @author Fabien Potencier <fabien@symfony.com>
+ * @author Jan Schädlich <jan.schaedlich@sensiolabs.de>
  *
  * @experimental in 5.0
  */
-class Recipient
+class Recipient implements EmailRecipientInterface, SmsRecipientInterface
 {
-    private $email;
+    use EmailRecipientTrait;
+    use SmsRecipientTrait;
 
-    public function __construct(string $email = '')
+    public function __construct(string $email = '', string $phone = '')
```

Comment:
When setting the phone to an empty string, it will be passed anyway to SMS transports as it implements the interface. Before, we checked for a non-empty phone, which I think must be kept.

Reply:
I think keeping the non-empty phone check would mean if I forgot to set a phone number even though I would like to send a `SmsMessage`, the `SmsChannel` would be skipped and I don't get any information why my notification was not sent as sms.

I would suggest a different approach: We could make the creation of `SmsMessage` more strict and throw an `InvalidArgumentException` with a proper exception message in case the `$phone` argument is empty. We could apply the same to `EmailMessage`

Reply:
@fabpot WDYT?

Reply:
Works for me.

Reply:
I think you forgot to enforce a non-empty phone on SmsMessage (and the same for the other messages), right?

Reply:
Oh no 🙈 thats true. On it.

Reply:
Fixed: d11567f8d37d6423facd4f12ccd9736345859fc7